python_modules/dagster/dagster/_core/execution/plan/external_step.py

```diff
@@ -236,6 +237,21 @@ def run_step_from_ref(
     check.inst_param(instance, "instance", DagsterInstance)
     step_context = step_run_ref_to_step_context(step_run_ref, instance)
 
+    if step_context.has_partition_key:
```

Comment:
The only bit of feedback I have is to leave a note here along the lines of:

```
Note: This is a patch that enables using DynamicPartitionsDefinitions with step launchers in the specific case where...

1. The external step operates on a single dynamic partition.
2. No dynamic partitions are added in this external step.

A more complete solution would require including all dynamic partitions on the StepRunRef object.
```

Happy to merge this in once added!

Reply:
@OwenKephart thanks for the feedback - its done! 👍